token-metadata/program/src/processor/metadata/print.rs

```diff
@@ -0,0 +1,194 @@
+    // let new_edition_account_info = next_account_info(account_info_iter)?;
+    // let master_edition_account_info = next_account_info(account_info_iter)?;
+    // let mint_info = next_account_info(account_info_iter)?;
+    // let token_record_info = next_account_info(account_info_iter)?;
+    // let edition_marker_info = next_account_info(account_info_iter)?;
+    // let mint_authority_info = next_account_info(account_info_iter)?;
+    // let payer_account_info = next_account_info(account_info_iter)?;
+    // let owner_account_info = next_account_info(account_info_iter)?;
+    // let token_account_info = next_account_info(account_info_iter)?;
+    // let update_authority_info = next_account_info(account_info_iter)?;
+    // let master_metadata_account_info = next_account_info(account_info_iter)?;
+    // let token_program_account_info = next_account_info(account_info_iter)?;
+    // let system_account_info = next_account_info(account_info_iter)?;
+
+    // CHECK: Checked in process_mint_new_edition_from_master_edition_via_token_logic
+    let edition_metadata_info = ctx.accounts.edition_metadata_info;
+    // CHECK: Checked in process_mint_new_edition_from_master_edition_via_token_logic
+    let edition_account_info = ctx.accounts.edition_info;
+    // CHECK: Checked in process_mint_new_edition_from_master_edition_via_token_logic
+    let edition_mint_info = ctx.accounts.edition_mint_info;
+    let edition_token_account_owner_info = ctx.accounts.edition_token_account_owner_info;
+    let edition_token_account_info = ctx.accounts.edition_token_account_info;
+    // CHECK: Checked in process_mint_new_edition_from_master_edition_via_token_logic
+    let edition_mint_authority_info = ctx.accounts.edition_mint_authority_info;
+    let edition_token_record_info = ctx.accounts.edition_token_record_info;
+    // CHECK: Checked in process_mint_new_edition_from_master_edition_via_token_logic
+    let master_edition_info = ctx.accounts.master_edition_info;
+    // CHECK: Checked in process_mint_new_edition_from_master_edition_via_token_logic
+    let edition_marker_pda_info = ctx.accounts.edition_marker_pda_info;
+    // CHECK: Checked in process_mint_new_edition_from_master_edition_via_token_logic
+    let payer_info = ctx.accounts.payer_info;
+    // CHECK: Checked in process_mint_new_edition_from_master_edition_via_token_logic
+    let master_token_account_owner_info = ctx.accounts.master_token_account_owner_info;
+    // CHECK: Checked in process_mint_new_edition_from_master_edition_via_token_logic
+    let master_token_account_info = ctx.accounts.master_token_account_info;
+    // CHECK: Checked in process_mint_new_edition_from_master_edition_via_token_logic
+    let master_metadata_info = ctx.accounts.master_metadata_info;
+    // CHECK: Checked in process_mint_new_edition_from_master_edition_via_token_logic
+    let update_authority_info = ctx.accounts.update_authority_info;
```

Comment:
I need to study this more but I don't see where this is checked in `process_mint_new_edition_from_master_edition_via_token_logic` and then I wonder why it can't just be gotten from `master_metadata_info` when we are printing.  Like, you don't want to specify and thus be able to change the update authority when printing right?

Reply:
It's used deep in the bowels of the print code. I originally didn't have this because we decided to get rid of the ability to set the UA on print but it seemed like it'd require a significant refactor to remove. I'll try to find the usage.

Reply:
Per offline discussion, just for record in this PR, we previously decided that with the new unified `Print` instruction that the creator should retain UA.  Since in the depths of the code it checks `is_signer` imo you could leave the account and just add a check in the code that `update_authority_info.key == master_metadata.update_authority`